SSA/Projects/FullyHomomorphicEncryption/Rewrites.lean

```diff
@@ -72,3 +72,79 @@ noncomputable def p1 : PeepholeRewrite (Op q n) [.polynomialLike, .polynomialLik
     }
 
 end ExampleComm
+
+section ExampleModulo
+
+@[irreducible]
+def irreduciblePow (q n : Nat) : Nat := q^n
+infixl:50 "**" => irreduciblePow
+
+
+variable {q : Nat} {n : Nat} [hq : Fact (q > 1)]
+
+-- code generator does not support recursor 'Decidable.rec' yet, consider using 'match ... with' and/or structural recursion
```

Comment:
What does this comment refer to?

Reply:
It explains why the code is marked `noncomputable`. The `Decidable.rec` is from the `poly.const`, IIRC.

Reply:
Addressed, I changed the comment.